src/plugins/data_view_editor/public/service.ts

```diff
@@ -0,0 +1,177 @@
+/*
```

Comment:
nit: I would suggest renaming the file name to `data_view_editor_service`. Otherwise harder to look up in IDE (IMHO)